### PR TITLE
[IASV-12] Integrate API SiciliaVola

### PR DESCRIPTION
**Pull request by debiff**
This PR integrates the API needed to manage the IO app SiciliaVola flow. 

In order to maintain the given order, the API are been separated in `secured`, `unsecured` and `auth`.

Note that the swagger `assets/SiciliaVola.yml` it's temporary and will be substituted when will be publicated.

### CODE REVIEW

File: src/routers/features/siciliaVola/secured/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      // validate the body value
+      res.sendStatus(500);
+      return;
+    }
+
+    if (vouchersBeneficiary.listaRisultati) {
+      vouchersBeneficiary = {
+        ...vouchersBeneficiary,
+        listaRisultati: vouchersBeneficiary.listaRisultati.filter(
+          v => v.idVoucher !== voucherId
```
Comment (Undermaken) [inline src/routers/features/siciliaVola/secured/index.ts:70]: Maybe this comparison has no sense: `number | undefined` !== `any` (the body)
Perhaps would you compare two numbers?

```typescript
addHandler(
  securedSvRouter,
  "get",
  addPrefix("/beneficiario/annullaVoucher"),
  (req, res) => {
    const { voucherId } = req.body;
    const maybeVoucherId = t.Integer.decode(voucherId);
    if (maybeVoucherId.isLeft()) {
      // validate the body value
      res.sendStatus(500);
      return;
    }

    if (vouchersBeneficiary.listaRisultati) {
      vouchersBeneficiary = {
        ...vouchersBeneficiary,
        listaRisultati: vouchersBeneficiary.listaRisultati.filter(
          v => v.idVoucher !== maybeVoucherId.value
        )
      };
    }
    res.sendStatus(200);
    return;
  }
);
```

File: src/server.ts [hunk 2]
```diff
@@ -70,7 +73,10 @@ app.use(
   cgnRouter,
   cgnMerchantsRouter,
   cgnGeoRouter,
-  euCovidCertRouter
+  euCovidCertRouter,
+  unsecuredSvRouter,
+  securedSvRouter,
+  authSvRouter
```
Comment (Undermaken) [inline src/server.ts:79]: I don't know if it works, but I guess a router can use other routers. If this works, you could export only a router that includes the 3 specific one

something like
```typescript
export const siciliaVolaRouter = Router();
siciliaVolaRouter.use([unsecuredSvRouter, securedSvRouter, authSvRouter]);
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: src/routers/features/siciliaVola/unsecured/index.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * Get the province list given a RegionID
+ */
+addHandler(
+  unsecuredSvRouter,
+  "get",
+  addPrefix("/province/:region_id"),
+  (req, res) => {
+    const regionId = req.params.region_id;
+
+    const regions = readFileAsJSON(
```
[the remark this answers is not in the document]
Comment (debiff) [reply src/routers/features/siciliaVola/unsecured/index.ts]: Thank you! Modified in all the APIs in [ebc4230](https://github.com/pagopa/io-dev-api-server/pull/60/commits/ebc42306499741868fe8043f13873da7443474c4)

File: src/server.ts (hunk 2 above)
Comment (debiff) [reply src/server.ts]: Modified in [4bcaa71](https://github.com/pagopa/io-dev-api-server/pull/60/commits/4bcaa711a0584dc7936247f19133bdfff137d435)

File: src/routers/features/siciliaVola/secured/index.ts (hunk 1 above)
Comment (debiff) [reply src/routers/features/siciliaVola/secured/index.ts]: Yes, you're right it's a bug. Fixed in [eab0296](https://github.com/pagopa/io-dev-api-server/pull/60/commits/eab0296bb2b82a03cb5d2275a1b6a944f6fc6fae)

File: src/routers/features/siciliaVola/auth.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export const authSvRouter = Router();
+
+const addPrefix = (path: string) => addApiV1Prefix(`/mitvoucher/auth${path}`);
+
+/**
+ * Get the mit auth token
+ */
+addHandler(securedSvRouter, "get", addPrefix("/token"), (_, res) =>
+  res.json({ token: faker.random.uuid() })
```
Comment (Undermaken) [inline src/routers/features/siciliaVola/auth.ts:15]:
```suggestion
  res.json({ token: faker.datatype.uuid() })
```
this is deprecated ([I realized it just minutes ago](https://github.com/pagopa/io-dev-api-server/commit/579e19e6864e1df01ce69fcaa5872c888b18afd9#))

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: src/routers/features/siciliaVola/auth.ts (hunk 4 above)
Comment (Undermaken) [reply src/routers/features/siciliaVola/auth.ts:15]: @debiff fyi
I committed this change

**Review by Undermaken — APPROVED** [review]

**Pull request merged**